### PR TITLE
Update example with zio-config

### PR DESCRIPTION
**Pull request by mvelimir**

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -52,6 +52,9 @@ object Dependencies {
     "com.github.pureconfig"        %% "pureconfig"                    % "0.16.0",
     "com.softwaremill.sttp.client" %% "async-http-client-backend-zio" % Versions.sttp,
     "com.softwaremill.sttp.client" %% "circe"                         % Versions.sttp,
+    "dev.zio"                      %% "zio-config"                    % "1.0.6",
```
Comment (drmarjanovic) [inline project/Dependencies.scala]: Should we extract `1.0.6` to `Versions`?

File: opentracing-example/src/main/scala/zio/telemetry/opentracing/example/config/Configuration.scala [hunk 2]
```diff
@@ -13,11 +14,14 @@ object Configuration {
   }
 
   object Live extends Service {
-    val load: Task[Config] = Task.effect(ConfigSource.default.loadOrThrow[Config])
+    val load: Task[Config] = for {
+      source <- ZIO.fromEither(TypesafeConfigSource
+        .fromHoconFile(new java.io.File("opentracing-example/src/main/resources/application.conf")))
```
Comment (drmarjanovic) [inline opentracing-example/src/main/scala/zio/telemetry/opentracing/example/config/Configuration.scala]: Can we avoid this hard-coded path?

Comment (mijicd) [reply opentracing-example/src/main/scala/zio/telemetry/opentracing/example/config/Configuration.scala]: Don't do this, zio-config and typesafe config can handle the default path easily. Take a look [here](https://github.com/zio/zio-redis/blob/master/example/src/main/scala/example/Main.scala#L23). Also, please do the following:

- rename `Config` to `AppConfig`
- try to remove `Configuration` (i.e. load from main, see zio-redis)
- extract  configuration classes from companion object into their own files

File: project/Dependencies.scala (hunk 1 above)
Comment (mijicd) [reply project/Dependencies.scala]: Definitely.

**Review by mijicd — CHANGES_REQUESTED** [review]

File: project/Dependencies.scala (hunk 1 above)
Comment (mvelimir) [reply project/Dependencies.scala]: Done.

File: opentracing-example/src/main/scala/zio/telemetry/opentracing/example/config/Configuration.scala (hunk 2 above)
Comment (mvelimir) [reply opentracing-example/src/main/scala/zio/telemetry/opentracing/example/config/Configuration.scala]: Done.

**Review by drmarjanovic — APPROVED** [review]

**Review by miloradvojnovic — APPROVED** [review]

**Review by mijicd — APPROVED** [review]

**Pull request merged**